src/pages/Admin/Charity/Permissions/useSubmit.ts

```diff
@@ -0,0 +1,63 @@
+import { useFormContext } from "react-hook-form";
+import { ProposalMeta } from "pages/Admin/types";
+import { useAdminResources } from "pages/Admin/Guard";
+import { useErrorContext } from "contexts/ErrorContext";
+import { useGetWallet } from "contexts/WalletContext";
+import CW3 from "contracts/CW3";
+import SettingsController from "contracts/SettingsController";
+import useCosmosTxSender from "hooks/useCosmosTxSender";
+import { isEmpty } from "helpers";
+import { getPayloadDiff } from "helpers/admin";
+import createUpdateEndowmentControllerMsg from "./createUpdateEndowmentControllerMsg";
+import { FormValues } from "./schema";
+
+export default function useSubmit() {
+  const { id, cw3, propMeta, endow_type } = useAdminResources<"charity">();
+  const { handleError } = useErrorContext();
+  const {
+    formState: { isSubmitting },
+    handleSubmit,
+    reset,
+  } = useFormContext<FormValues>();
+  const { wallet } = useGetWallet();
+  const sendTx = useCosmosTxSender();
+
+  async function onSubmit({ initialValues, ...newValues }: FormValues) {
+    try {
+      const diff = getPayloadDiff(initialValues, newValues);
+
+      if (isEmpty(Object.entries(diff))) {
+        return handleError("No changes detected");
+      }
+
+      const settingsController = new SettingsController(wallet);
+      const msg = createUpdateEndowmentControllerMsg(id, diff);
+      const embeddedMsg =
+        settingsController.createEmbeddedUpdateEndowmentControllerMsg(msg);
+
+      const meta: ProposalMeta = { type: "endow_controller" };
+
+      const adminContract = new CW3(wallet, cw3);
+      const proposalMsg = adminContract.createProposalMsg(
+        "Update permission settings",
+        `Update permission settings for endowment id:${id}`,
+        [embeddedMsg],
+        JSON.stringify(meta)
+      );
+
+      await sendTx({
+        msgs: [proposalMsg],
+        ...propMeta,
+      });
+    } catch (error) {
+      handleError(error);
+    }
+  }
+
+  return {
+    disabled:
```

Comment:
@misicnenad, @SovereignAndrey, @abbasali1994 

this might be a discussion on general handling of `disabled` state of a form

<img width="1453" alt="image" src="https://user-images.githubusercontent.com/89639563/224996341-0c227029-384a-4233-a63d-7bfee38a6137.png">

as a developer, when I go to this page and see everything disabled - I know what to do. 

* I need to connect my wallet
* I have to be on the right endowment  / must have the right authorization

but for someone who doesn't know that internals of this. It might be confusing to see why everything is `disabled` and user might try bunch of interactions just to make things `enabled`

Solution to this:
1. If we want everything disabled at first, we should provide helpful text/tooltips why the form is disabled like

or 
2. Just enable the form initially, and when user tries to submit be prompted with messages like
`Your wallet needs to be connected to submit this transaction`
`Your wallet is not authorized to modify settings` 
etc..

this is also considering admin list in wallet dropdown is not perfectly in 
<img width="418" alt="image" src="https://user-images.githubusercontent.com/89639563/224998408-39b077a4-6767-4bc7-9d89-02d71f412e74.png">
sync so a `user` might be on a page where he is not a member of

Reply:
@SovereignAndrey @ap-justin might be useful for disconnected/unauthorized users to have a top banner on each page with a tooltip stating `"Your wallet needs to be connected and authorized to modify settings"` (a combination of the two messages from @ap-justin's comment, but we could have separate messages ofc).
It could be similar to the one in the Withdraw page:
![image](https://user-images.githubusercontent.com/19427053/225217320-d033fb88-987f-4db7-89ce-32666bd255b2.png)

Reply:
> @SovereignAndrey @ap-justin might be useful for disconnected/unauthorized users to have a top banner on each page with a tooltip stating `"Your wallet needs to be connected and authorized to modify settings"` (a combination of the two messages from @ap-justin's comment, but we could have separate messages ofc). It could be similar to the one in the Withdraw page: ![image](https://user-images.githubusercontent.com/19427053/225217320-d033fb88-987f-4db7-89ce-32666bd255b2.png)

That would be good. It's also a bigger issue than just on this page alone for how we ought to display forms to non-connected wallet users. Ideally, we have separate, non-form based views for when wallets are not connected.

Reply:
@SovereignAndrey @ap-justin since this is a problem related to _all_ admin pages, it's safe to say that this requirement is out of scope of this PR and should be tackled separately.

Reply:
@SovereignAndrey, @misicnenad 
> since this is a problem related to all admin pages,

currently, all forms in `pages/Admin` uses the `2`nd approach 
that is: forms are submittable - then prompts user on problems

sample:
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/89639563/225824609-5b60e236-417f-46d9-baa7-989abd4eb4f6.png">

where problems manifest in this order
1. submitting no changes
2. wallet is not connected
3. unauthorized ( admin membership check)

now  we have 
4. permissions check for select (not all ) pages 

<img width="351" alt="image" src="https://user-images.githubusercontent.com/89639563/225825139-e52f2ca6-ab31-4a30-8a43-558b4942e510.png">


starting with the general
`{ update_controller_settings }` 

we could simply add a simple check in the submit of this page rather than disabling everything

that is , add this simple line

```
 await sendTx({
        //...
        isAuthorized: endow_controller.owner_controlled & propMeta.isAuthorized || endow_controller?.delegate?.address === wallet.address
      });
```

and on a separate PRs say
in `edit_profile`

```
 await sendTx({
        //...
        isAuthorized: name.owner_controlled & propMeta.isAuthorized || endow_controller?.delegate?.address === wallet.address
      });
```

Reply:
> currently, all forms in pages/Admin uses the 2nd approach
that is: forms are submittable - then prompts user on problems

@SovereignAndrey @ap-justin even though it's used on every other admin page, I would argue it is still bad UX and shouldn't be introduced here.
If a user is unable to change and submit anything on the page, that "thing" should be disabled for them to indicate so.

Reply:
@misicnenad, How can you be certain that this is bad UX ? 
>  I would argue it is still bad UX and shouldn't be introduced here.

this is to mitigate this, which was introduced in this PR
> but for someone who doesn't know that internals of this. It might be confusing to see why everything is disabled and user might try bunch of interactions just to make things enabled

by literally adding just this simple line in `submit`
```
isAuthorized: endow_controller.owner_controlled & propMeta.isAuthorized || endow_controller?.delegate?.address === wallet.address
```

Reply:
@ap-justin 
> How can you be certain that this is bad UX 

Quoting my [last comment](https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1885#discussion_r1141664988):
> If a user is unable to change and submit anything on the page, that "thing" should be disabled for them to indicate as much.

> but for someone who doesn't know that internals of this. It might be confusing to see why everything is disabled and user might try bunch of interactions just to make things enabled

It would be confusing only if we do not add the top banner tooltip, read [my comment on this](https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1885#discussion_r1136558155) and [Andrey's response](https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1885#discussion_r1136627773): 

> > @SovereignAndrey @ap-justin might be useful for disconnected/unauthorized users to have a top banner on each page with a tooltip stating `"Your wallet needs to be connected and authorized to modify settings"` (a combination of the two messages from @ap-justin's comment, but we could have separate messages ofc). It could be similar to the one in the Withdraw page: ![image](https://user-images.githubusercontent.com/19427053/225217320-d033fb88-987f-4db7-89ce-32666bd255b2.png)
> 
> That would be good. It's also a bigger issue than just on this page alone for how we ought to display forms to non-connected wallet users. Ideally, we have separate, non-form based views for when wallets are not connected in this admin area where the pages are publicly viewable and then also something like the above mentioned notice message to hint at other functionality if a wallet is connected.

We can leave things as-is for now as this is not and will not be in production yet, the tooltip banner can be created before it is pushed there @SovereignAndrey